File: Gobackn.cpp
#include <stdio.h>
int main(){
    int window = 0;
    printf("enter Window size : ");
    scanf("%d", &window);
    int sent=0, ack, i = 0;
    while (1)
    {
        for (i = 0; i < window; i++)
        {
            printf("frame Transmitted %d \n", sent);
            sent++;
            if (sent == window)
            {
                break;
            }
        }
        printf("enter last received acknowledgment : ");
        scanf("%d", &ack);
        if (ack == window)
        {
            break;
        }
        else
        {
            sent = ack;
        }
    }
}

File: cn/Gobackn.java
import java.util.Scanner;

public class Gobackn {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Enter Window size: ");
        int window = scanner.nextInt();
        
        int sent = 0;
        int ack;
        
        while (true) {
            for (int i = 0; i < window; i++) {
                System.out.println("Frame Transmitted " + sent);
                sent++;
                if (sent == window) {
                    break;
                }
            }
            
            System.out.print("Enter last received acknowledgment: ");
            ack = scanner.nextInt();
            
            if (ack == window) {
                break;
            } else {
                sent = ack;
            }
        }
        
        scanner.close();
    }
}
